### PR TITLE
gruvbox-plus-icon-pack: init at unstable-2023-11-07

### LINKED ISSUE #266170: Package request: `gruvbox-plus-icon-pack`
**Project description**
Gruvbox Plus icon pack for Linux desktops based on Gruvbox color theme.

**Metadata**

* homepage URL: https://github.com/SylEleuth/gruvbox-plus-icon-pack
* source URL: https://github.com/SylEleuth/gruvbox-plus-icon-pack
* license: GPL-3.0
* platforms: linux

### PR DESCRIPTION
**Pull request by eureka-cpu**
## Description of changes
Closes #266170 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/icons/gruvbox-plus-icon-pack/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  '';
+
+  dontDropIconThemeCache = true;
+
+  meta = with lib; {
+    description = "Gruvbox Plus icon pack for Linux desktops based on Gruvbox color theme.";
+    homepage = "https://github.com/SylEleuth/gruvbox-plus-icon-pack";
+    license = licenses.gpl3Only;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.eureka-cpu ];
```
Comment (IogaMaster) [inline pkgs/data/icons/gruvbox-plus-icon-pack/default.nix:32]:
```suggestion
    maintainers = with maintainers; [ eureka-cpu ];
```

File: pkgs/data/icons/gruvbox-plus-icon-pack/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{ lib, stdenvNoCC, fetchFromGitHub, gtk3, breeze-icons, gnome-icon-theme, hicolor-icon-theme }:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "gruvbox-plus-icon-pack";
+  version = "4.0";
+
+  src = fetchFromGitHub {
+    owner = "SylEleuth";
+    repo = pname;
+    rev = "refs/tags/v${version}";
```
Comment (IogaMaster) [inline pkgs/data/icons/gruvbox-plus-icon-pack/default.nix:10]:
```suggestion
    rev = "v${version}";
```
This works fine as is, just a style choice. It is more common not to use `refs/tags`

**Review by IogaMaster — CHANGES_REQUESTED** [review]
I told eureka to try to use `dontBuild` and `dontConfigure`. It's not in the review, because I am not sure if it will work or not.

File: pkgs/data/icons/gruvbox-plus-icon-pack/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    sha256 = "sha256-KefCHHFtuh2wAGBq6hZr9DpuJ0W99ueh8i1K3tohgG8=";
+  };
+
+  nativeBuildInputs = [ gtk3 ];
+
+  propagatedBuildInputs = [ breeze-icons gnome-icon-theme hicolor-icon-theme ];
+
```
Comment (IogaMaster) [inline pkgs/data/icons/gruvbox-plus-icon-pack/default.nix:17]:
```suggestion
dontBuild = true;
dontConfigure = true;
```

File: pkgs/data/icons/gruvbox-plus-icon-pack/default.nix (hunk 2 above)
Comment (eureka-cpu) [reply pkgs/data/icons/gruvbox-plus-icon-pack/default.nix:10]: This actually was causing the build to fail for me

**Review by IogaMaster — CHANGES_REQUESTED** [review]

File: pkgs/data/icons/gruvbox-plus-icon-pack/default.nix (hunk 2 above)
Comment (eureka-cpu) [reply pkgs/data/icons/gruvbox-plus-icon-pack/default.nix]: wasn't working bc of a typo, went ahead and updated

Comment (IogaMaster) [reply pkgs/data/icons/gruvbox-plus-icon-pack/default.nix]: This was because of a typo in the package previously, so removing should work!

**Comment by iynaix** [comment]
Result of `nixpkgs-review pr 266188` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gruvbox-plus-icon-pack</li>
  </ul>
</details>

**Review by IogaMaster — APPROVED** [review]
Looks great!
Thanks for contributing to nixpkgs! Keep up the good work!

**Comment by romildo** [comment]
Make a separate commit for adding the user to the maintainer list.

**Comment by RGBCube** [comment]
Is the breeze-icons necessary? gnome icons are already used

**Comment by romildo** [comment]
> Is the breeze-icons necessary? gnome icons are already used

Not strictly. The `index.theme` file has the line:
```
Inherits=Breeze Dark, gnome, hicolor
```
That means that icons not provided by this theme will be taken first from Breeze Dark, then from gnome, then from hicolor, in that order. In other words, this theme inherits from them. Therefore those themes complement this one. If the parent themes are not installed, then some icons may not be found.

Then, for completeness of the icon set, we use to install also the parent themes.

**Comment by RGBCube** [comment]
> > Is the breeze-icons necessary? gnome icons are already used
> 
> Not strictly. The `index.theme` file has the line:
> 
> ```
> Inherits=Breeze Dark, gnome, hicolor
> ```
> 
> That means that icons not provided by this theme will be taken first from Breeze Dark, then from gnome, then from hicolor, in that order. In other words, this theme inherits from them. Therefore those themes complement this one. If the parent themes are not installed, then some icons may not be found.
> 
> Then, for completeness of the icon set, we use to install also the parent themes.

Ah I see! thats fine then, only blockers to this PR is the changes suggested now then

**Comment by eureka-cpu** [comment]
@RGBCube would you like me to add you as a maintainer as well?

**Comment by RGBCube** [comment]
> @RGBCube would you like me to add you as a maintainer as well?

Sure

**Comment by RGBCube** [comment]
Could you add push perms to me on your fork so I could apply the code review?

**Comment by RGBCube** [comment]
Nevermind, it doesn't build, fixing rn.

**Comment by RGBCube** [comment]
Could this be merged?

**Pull request merged**